src/API/AbstractAPIClient.php

```diff
@@ -39,42 +39,74 @@ public function callAPI(Request $request)
 
             $request = $this->beforeSend($request);
 
+            $method = $request->getMethod();
+            $params = $request->getParameters();
+            $url = $request->getUrl();
             $headers = $request->getHeaders();
-            $contentTypeHeader = $headers[self::CONTENT_TYPE_KEY];
-            $bodyType = (($contentTypeHeader === self::APPLICATION_JSON_KEY) ? 'json' : 'body');
-
-            $requestOptions = array(
-                'headers' => $request->getHeaders(),
-                'query' => $request->getParameters(),
-                $bodyType => $request->getBody(),
-            );
+            $body = $request->getBody();
 
-            if ($this->config->getValue('debug')) {
-                $requestOptions['debug'] = fopen('php://stderr', 'w');
+            $isPaginatable = false;
+            if ($method === 'GET') {
+                $isPaginatable = true;
             }
 
-            $apiRequest = $client->createRequest($request->getMethod(), $request->getUrl(), $request->getHeaders(), $request->getBody(), $request->getParameters());
+            $mergedResponse = null;
 
-            $method = $request->getMethod();
+            $currentPage = 1;
+            $totalPages = 1;
 
-            // Since Guzzle automatically overwrites a new header when the request
-            // is POST / PATCH / DELETE / PUT, we need to overwrite the Content-Type header
-            // with setBody() function.
-            if ($method !== 'GET') {
-                $apiRequest->setBody(json_encode($request->getBody()), 'application/json');
-            }
+            while ($totalPages >= $currentPage) {
```

Comment:
Move this while loop into a function that takes a request with `result_info` and returns the merged result.